### ISSUE TITLE
web-mode-comment-formats doesn't seem to work for jsx files after update

### ISSUE THREAD

**Issue by jacktrades** [fxbois/web-mode#774]
After many months, I've updated to 20160717.1201 and I noticed that commenting js code in jsx files uses the block comment syntax `/* */` instead of the line comment syntax `//` I've set with `web-mode-comment-formats`.

```
(add-to-list 'web-mode-comment-formats '("jsx" . "// "))
```

The line comments work fine in .js files, so it seems to be a jsx only issue.

**Comment by strom** [comment]
I can't get `\\` comments to work, period. It's very frustrating.

**Comment by strom** [comment]
20170830.1210 seems to have fixed it

**Comment by jacktrades** [comment]
@fxbois this still doesn't seem to be working for me in `jsx` files.  I just updated to the latest version from melpa: 20180217.916

My settings are:

```elisp
(add-to-list 'web-mode-comment-formats '("javascript" . "// "))
(add-to-list 'web-mode-comment-formats '("jsx" . "// "))
```

And `describe-variable` for `web-mode-comment-formats` in a `jsx` buffer is:

```
web-mode-comment-formats is a variable defined in `web-mode.el'.
Its value is (("jsx" . "// ")
 ("javascript" . "// ")
 ("java" . "/*")
 ("javascript" . "/*")
 ("php" . "/*")
 ("css" . "/*"))

Local in buffer main.jsx; global value is the same.
```

**Comment by fxbois** [comment]
@jacktrades should be fixed with last commit

**Comment by jacktrades** [comment]
@fxbois thanks!

Actually, it still didn't work given the settings I posted.  I checked the commit that fixes this though (72b5e9d6f8997ccafefb038578a25d088150d369) and realized that it's because I have a trailing space in my settings (`"// "` vs `"//"`).  Once I updated that it works!

One thing I noticed is that commenting a block with a blank line at the end introduces some trailing whitespace.  Though maybe this should be a separate issue, since it looks like it is existing behavior.

![screen shot 2018-03-06 at 6 39 40 am](https://user-images.githubusercontent.com/385989/37032864-8b0ca3e2-2109-11e8-9711-ebd27213fb07.png)
